midend/local_copyprop.cpp

```diff
@@ -640,7 +640,8 @@ void DoLocalCopyPropagation::apply_table(DoLocalCopyPropagation::TableInfo *tbl)
     for (auto key : tbl->keyreads) {
         forOverlapAvail(key, [&remaps_seen, key, tbl, this](cstring vname, VarInfo *var) {
             remaps_seen.insert(vname);
-            if (var->val && lvalue_out(var->val)->is<IR::PathExpression>()) {
+            if (var->val && (lvalue_out(var->val)->is<IR::PathExpression>() ||
+                             lvalue_out(var->val)->is<IR::MethodCallExpression>())) {
```

Comment:
This will propagate method calls that can have side-effects.
That may be problematic.
isValid() is really a special case.

Reply:
I modified the PR to prevent method calls with side effect to propagate into table keys.